src/common/com/intellij/plugins/haxe/lang/psi/HaxeMetaContainerElement.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2018 Ilya Malanin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.lang.psi;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public interface HaxeMetaContainerElement extends HaxePsiCompositeElement {
```

Comment:
It will not work I guess, this kind of stuff is truly hard to support by grammar-kit, right now I'm working on custom lexer and parser for Haxe code (kind of experiment)

Reply:
I can make the meta "disappear" from the parsing rules and still leave it in the PSI for you to find.

I've thought about writing my own parser more than a few times.  You might take a look at Simon's menhir-based parser (menhir is like grammar-kit for ocaml).  It parses most of the haxe language.  He got stuck at pre-processing.  Maybe we should pick that up, port it to Haxe (or grammar-kit) and finish.  That code is at https://github.com/vshaxe/hxparser.

I'd really like to use the compiler's parser, and that seems in reach at the moment because of the new Haxe-in-Haxe effort.  When that is finished (and it's going really fast), I think we will be able to cross-compile to Java and then we might be able to use it directly in the plugin.  Except that Simon told me that would be prohibitive because the current Haxe parser throws tokens out as it processes and refactoring is going to be really hard.

Reply:
Issue #757